rebasehelper/checkers/licensecheck_tool.py

```diff
@@ -0,0 +1,180 @@
+        new_sources_path = os.path.join(cls.workspace_dir, 'new_sources')
+
+        diffs = []
+        for source_dir in [old_sources_path, new_sources_path]:
+            out = six.StringIO()
+            ProcessHelper.run_subprocess(["/usr/bin/licensecheck", source_dir, "--machine", "--recursive"],
+                                         output_file=out)
+            diff = {}
+            for l in out:
+                # licensecheck output format: 'Filepath\tlicense'
+                file_path, dlicense = l.split('\t')
+                # Get the important file path part. Strip ".../new_sources/package-{version}/..." part away
+                file_path = os.path.relpath(file_path, source_dir)
+                diff[file_path] = dlicense.strip()
+            diffs.append(diff)
+
+        old_lics, new_lics = set(), set()
+        changes = {'added': {}, 'changed': {}, 'removed': {}}
+        # Get changed licenses in existing files
+        for new_file, new_license in six.iteritems(diffs[1]):
+            new_lics.add(new_license)
+            for old_file, old_license in six.iteritems(diffs[0]):
+                old_lics.add(old_license)
+
+                if (new_file == old_file and
+                   (new_license != old_license)):
+                    new_key = '{} => {}'.format(old_license, new_license)
+                    if new_license == 'UNKNOWN':
+                        if old_license not in changes['removed']:
+                            changes['removed'][old_license] = []
+                        changes['removed'][old_license].append(new_file)
+                    elif old_license == 'UNKNOWN':
+                        if new_license not in changes['added']:
+                            changes['added'][new_license] = []
+                        changes['added'][new_license].append(new_file)
+                    else:
+                        if new_key not in changes['changed']:
+                            changes['changed'][new_key] = []
+                        if new_file not in changes['changed'][new_key]:
+                            changes['changed'][new_key].append(new_file)
+
+        # Get newly appeared files
+        for new_file, new_license in six.iteritems(diffs[1]):
+            if new_file not in diffs[0]:
+                if new_license not in changes['added']:
+                    changes['added'][new_license] = []
+                changes['added'][new_license].append(new_file)
+
+        # Get removed appeared files
+        for old_file, old_license in six.iteritems(diffs[0]):
+            if old_file not in diffs[1]:
+                if old_license not in changes['removed']:
+                    changes['removed'][old_license] = []
+                changes['removed'][old_license].append(old_file)
+
+        new_licenses = []
+        for new in new_lics:
+            if new not in old_lics:
+                cls.license_changes = True
+                new_licenses.append(new)
```

Comment:
You should also check whether a license disappeared - that equals to license change too.

Reply:
You are right, and it still does not count with packages that are not in a directory after unpacking the tarball.

Reply:
I'm not sure I understand, what packages are not in which directory?

Reply:
While running tests licensecheck fails due to unexpected format of untarred archive `test-1.0.2.tar.xz` which does not contain a subdirectory `test-1.0.2`, just a file `text.txt`.
Licensecheck always tries to cut the first "version" directory off from the file names for more convenient manipulation. 
Most packages are made so, that the tarball contains the version subdirectory.

Do you think we should solve it somehow, or just move the testing file into a versioned subdirectory?

Reply:
I will probably solve it by try-catch with appending `../` to the exceptional files.

Reply:
Not sure which tests, but this is certainly a bug. The archive is not even being extracted in `%prep` of `test.spec`. `%setup` with `-c` option has to be used in `%prep` or top-level directory has to be added to the archive. And `Application.prepare_sources()` has to be modified to take potential `-c` option into account.

Reply:
Ok, I was wrong. `%setup -q -a 5` actually means to extract `Source0` and `Source5`. But the `-c` option or top-level dir in `Source0` archive is still missing. I don't think a package can be built without having top-level dir (created either way).

Reply:
So, to conclude this: the test has to be fixed, and source checkers should use paths returned from `Application.prepare_sources()`, which should be modified to take `%prep` into account.